evalml/tests/utils_tests/test_gen_utils.py

```diff
@@ -214,3 +215,28 @@ def test_check_random_state_equality():
     rs_1.set_state(tuple(['MT19937', np.array([1] * 624), 0, 1, 0.1]))
     rs_2.set_state(tuple(['MT19937', np.array([1] * 624), 1, 1, 0.1]))
     assert not check_random_state_equality(rs_1, rs_2)
+
+
+@patch('evalml.utils.gen_utils.import_or_raise')
+def test_jupyter_check_errors(mock_import_or_raise):
+    mock_import_or_raise.side_effect = ImportError
+    assert not jupyter_check()
+
+    mock_import_or_raise.side_effect = Exception
+    assert not jupyter_check()
+
+
+@patch('evalml.utils.gen_utils.import_or_raise')
+@patch('IPython.core.getipython.get_ipython')
+def test_jupyter_check(mock_ipython, mock_import_or_raise):
+    mock_import_or_raise.return_value = MagicMock()
+    mock_ipython.return_value = True
+    mock_import_or_raise.core.getipython.get_ipython().return_value = True
```

Comment:
I think this line needs to be changed to something like
```python
mock_import_or_raise.core.get_ipython.get_ipython = MagicMock()
mock_import_or_raise.core.get_ipython.get_ipython.return_value = True
```
At least I think that's on the right track, even if it doesn't work! Try digging into how to set attributes on mocks and that should get you on the right track.